src/protocol/libp2p/identify.rs

```diff
@@ -200,16 +201,31 @@ pub(crate) struct Identify {
 
 impl Identify {
     /// Create new [`Identify`] protocol.
-    pub(crate) fn new(
-        service: TransportService,
-        config: Config,
-        listen_addresses: Vec<Multiaddr>,
-    ) -> Self {
+    pub(crate) fn new(service: TransportService, config: Config) -> Self {
+        let listen_addresses = service.public_addresses();
+
+        let local_peer_id = service.local_peer_id();
+        let filtered_public_addr = config.public_addresses.into_iter().filter_map(|address| {
+            if address.is_empty() {
+                return None;
+            }
+            if let Some(peer_id) = PeerId::try_from_multiaddr(&address) {
+                if peer_id != local_peer_id {
+                    return None;
+                }
+
+                return Some(address);
+            }
+
+            Some(address.with(Protocol::P2p(local_peer_id.into())))
+        });
+        listen_addresses.inner.write().extend(filtered_public_addr);
```

Comment:
I don't think we should modify a global `libp2p` struct from Identify protocol.

Reply:
Yep that makes sense, have decided to keep the:
- user provided addresses
- listen addresses
- external addresses